x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -59,99 +59,86 @@ type InventoryMetricThresholdAllowedActionGroups = ActionGroupIdsOf<
 export const FIRED_ACTIONS_ID = 'metrics.inventory_threshold.fired';
 export const WARNING_ACTIONS_ID = 'metrics.inventory_threshold.warning';
 
-type InventoryThrehsoldActionGroup = typeof FIRED_ACTIONS_ID | typeof WARNING_ACTIONS_ID;
-
 export type InventoryMetricThresholdRuleTypeState = RuleTypeState; // no specific state used
 export type InventoryMetricThresholdAlertState = AlertState; // no specific state used
 export type InventoryMetricThresholdAlertContext = AlertContext; // no specific instance context used
 
-type InventoryMetricThresholdAlert = Alert<
-  InventoryMetricThresholdAlertState,
-  InventoryMetricThresholdAlertContext,
-  InventoryMetricThresholdAllowedActionGroups
->;
-type InventoryMetricThresholdAlertFactory = (
-  id: string,
-  reason: string,
-  actionGroup: InventoryThrehsoldActionGroup,
-  additionalContext?: AdditionalContext | null,
-  evaluationValues?: Array<number | null>,
-  thresholds?: Array<number | null>
-) => InventoryMetricThresholdAlert;
-
-export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =>
-  libs.metricsRules.createLifecycleRuleExecutor<
-    InventoryMetricThresholdParams & Record<string, unknown>,
-    InventoryMetricThresholdRuleTypeState,
-    InventoryMetricThresholdAlertState,
-    InventoryMetricThresholdAlertContext,
-    InventoryMetricThresholdAllowedActionGroups
-  >(
-    async ({
+export type InventoryMetricThresholdAlert = Omit<
+  ObservabilityMetricsAlert,
+  'kibana.alert.evaluation.values'
+> & {
+  // Defining a custom type for this because the schema generation script doesn't allow explicit null values
+  [ALERT_EVALUATION_VALUES]?: Array<number | null>;
+  [ALERT_EVALUATION_THRESHOLD]?: Array<number | null>;
```

Comment:
seems like THRESHOLD isn't omitted above ... I was able to add it to the `Omit<>` with no errors ...

Reply:
Fixed it, thanks.